doc/scripts/genrest.py

```diff
@@ -79,67 +270,86 @@ def expr_str(expr):
+        # Add an index entry for the symbol that links to its page. Also list
+        # its prompt(s), if any. (A symbol can have multiple prompts if it has
+        # multiple definitions.)
+        rst += "   * - :option:`CONFIG_{}`\n     - {}\n".format(
+            sym.name, " / ".join(node.prompt[0] for node in sym.nodes
+                                 if node.prompt))
 
-def main():
-    # Writes index.rst and the symbol RST files
+    if suffix is None:
+        fname = "index.rst"
+    else:
+        fname = "index-{}.rst".format(suffix)
+
+    write_if_updated(fname, rst)
 
-    if len(sys.argv) != 3:
-        sys.exit("usage: {} <Kconfig> <output directory>"
-                 .format(sys.argv[0]))
 
-    kconf = kconfiglib.Kconfig(sys.argv[1])
-    out_dir = sys.argv[2]
+def sym_index_header(title, link):
+    # write_sym_index_page() helper. Returns the RST for the beginning of a
+    # symbol index page, with title '<title> Configuration Options' and a link
+    # target 'configurations_options_<link>' pointing to the page. 'title'
+    # and 'link' can be None to skip the prefix/suffix on the title/link.
 
-    if os.environ.get("KCONFIG_TURBO_MODE") == "1":
-        write_dummy_index(kconf, out_dir)
+    if title is None:
+        title = "Configuration Options"
     else:
-        write_rst(kconf, out_dir)
+        title = title + " Configuration Options"
 
+    if link is None:
+        link = "configuration_options"
+    else:
+        link = "configuration_options_" + link
 
-def write_rst(kconf, out_dir):
-    # Generates all output pages
+    title += "\n" + len(title)*"="
 
-    # String with the RST for the index page
-    index_rst = INDEX_RST_HEADER
+    return """\
+.. _{}:
 
-    # Sort the symbols by name so that they end up in sorted order in index.rst
-    for sym in sorted(kconf.unique_defined_syms, key=attrgetter("name")):
-        # Write an RST file for the symbol
-        write_sym_rst(sym, out_dir)
+{}
 
-        # Add an index entry for the symbol that links to its RST file. Also
-        # list its prompt(s), if any. (A symbol can have multiple prompts if it
-        # has multiple definitions.)
-        index_rst += "   * - :option:`CONFIG_{}`\n     - {}\n".format(
-            sym.name,
-            " / ".join(node.prompt[0]
-                       for node in sym.nodes if node.prompt))
+Overview
+********
 
-    for choice in kconf.unique_choices:
-        # Write an RST file for the choice
-        write_choice_rst(choice, out_dir)
+:file:`Kconfig` files describe build-time configuration options (called symbols
+in Kconfig-speak), how they're grouped into menus and sub-menus, and
+dependencies between them that determine what configurations are valid.
 
-    write_if_updated(os.path.join(out_dir, "index.rst"), index_rst)
+:file:`Kconfig` files appear throughout the directory tree.
+:file:`subsys/power/Kconfig` might define power-related options, for example.
```

Comment:
Better would be:
```
For example, :file:`subsys/power/Kconfig` defines power-related options.
```

Reply:
Changed.

Renamed `--project-title` to `--non-module-title` too. Passing

    --modules ARM:arm:../arch/arm --non-module-title Core

gives the page below. Think it might make it clearer what it's for.

![Skärmbild från 2019-10-08 08-31-06](https://user-images.githubusercontent.com/154768/66372505-6d5a7b00-e9a6-11e9-9997-caf7565f56b3.png)